drivers/peci/peci_mchp_xec.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <drivers/peci.h>
+#include <soc.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(peci_mchp_xec, CONFIG_PECI_LOG_LEVEL);
+
+/* Maximum PECI core clock is the main clock 48Mhz */
+#define MAX_PECI_CORE_CLOCK 48000
+/* 1 ms */
+#define PECI_RESET_DELAY    1000
+/* 100 us */
+#define PECI_IDLE_DELAY     100
+/* 5 ms */
+#define PECI_IDLE_TIMEOUT   50
+/* 10 us */
+#define PECI_IO_DELAY       10
+
+struct peci_xec_config {
+	PECI_Type *base;
+	u8_t host_addr;
+	u8_t irq_num;
+};
+
+struct peci_xec_data {
+	struct k_sem tx_lock;
+};
+
+static int check_bus_idle(PECI_Type *base)
+{
+	u8_t delay_cnt = PECI_IDLE_TIMEOUT;
+
+	/* Wait until PECI bus becomes idle.
+	 * Note that when IDLE bit in the status register changes, HW do not
+	 * generate an interrupt, so need to poll.
+	 */
+	while (!(base->STATUS2 & MCHP_PECI_STS2_IDLE)) {
+		k_busy_wait(PECI_IDLE_DELAY);
+		delay_cnt--;
+
+		if (!delay_cnt) {
+			LOG_WRN("Bus is busy\n");
+			return -EBUSY;
+		}
+	}
+
+	return 0;
+}
+
+static int peci_xec_configure(struct device *dev, u32_t bitrate)
+{
+	const struct peci_xec_config *config = dev->config->config_info;
+	PECI_Type *base = config->base;
+	u16_t value;
+
+	/* Power down PECI interface */
+	base->CONTROL = MCHP_PECI_CTRL_PD;
+
+	/* Adjust bitrate */
+	value = MAX_PECI_CORE_CLOCK / bitrate;
+	base->OPT_BIT_TIME_LSB = value & MCHP_PECI_OPT_BT_LSB_MASK;
+	base->OPT_BIT_TIME_MSB = (value >> 8) & MCHP_PECI_OPT_BT_MSB_MASK;
```

Comment:
is there anything in the HAL that can allow you to removed the number 8

Reply:
No, added a macro into the driver